src/passes/hash-stringify-walker.cpp

```diff
@@ -79,15 +78,22 @@ HashStringifyWalker::makeRelative(uint32_t idx) const {
   // The upper_bound function returns an iterator to the first value in the set
   // that is true for idx < value. We subtract one from this returned value to
   // tell us which function actually contains the the idx
-  uint32_t funcStartIdx = *--funcIndices.upper_bound(idx);
-  return {idx - funcStartIdx, idxToFuncName.at(funcStartIdx)};
+  auto itr = *--idxToFuncName.upper_bound(idx);
+  return {idx - itr.first, itr.second};
 }
 
 std::vector<SuffixTree::RepeatedSubstring>
 StringifyProcessor::repeatSubstrings(std::vector<uint32_t>& hashString) {
   SuffixTree st(hashString);
-  std::vector<SuffixTree::RepeatedSubstring> substrings =
-    std::vector(st.begin(), st.end());
+  std::vector<SuffixTree::RepeatedSubstring> substrings(st.begin(), st.end());
+  for (auto substring : substrings) {
```

Comment:
It would be good to add a comment to this sort as well. Maybe something like `// Sort by increasing start index to ensure determinism.`.

Reply:
done